dacScanComparison2.py

```diff
@@ -0,0 +1,77 @@
+#
+#
+# This macro takes the name of the chamber and all the scandates from the scandates list file and produces tables for the twelve parameter of the VFAT electronics
+#
+#
+
+def getcolumn(filename, column, vfat):
+    f=open(filename,"r")
+    lines=f.readlines()
+    values=[]
+    if(vfat==-1):
+        for line in lines:
+            values.append(line.strip().split(' ')[column])
+    else:
+        values = "N.A."
+        for line in lines:
+            if(line.strip().split('\t')[0]==vfat):
+                values = line.strip().split('\t')[column]
+    f.close()
+    #print values
+    return values
+
+import argparse
+import os,commands
+import sys
+usage = 'python dacScanComparison.py python dacScanComparison.py /path/to/DACscans/ --sdlist scandatelist.txt --vfat all -o /path/for/the/results/'
+parser = argparse.ArgumentParser(usage)
+parser.add_argument('filepath', type=str, help="File path from which input information is read until the folder with the chamber name", metavar='filepath')
+parser.add_argument('--vfat', dest='vfat', type=str, default = 'none', help='')
+parser.add_argument('--sdlist', dest='sdlist', type=str, default = '', help='Names of the scandate file to analyse')
+parser.add_argument('-o','--outpath', dest='outpath', type=str, default = './', help='Path on the folder where the summary table will be written')
+parser.add_argument('--popup', dest='popup', default=False, action='store_true') 
+args = parser.parse_args()
+
+print("Analyzing: '%s'" %args.filepath)
+print("scandatelistfile: '%s'" %args.sdlist)
+
+chamber = getcolumn(args.sdlist,0,-1)[0]
+scandates = getcolumn(args.sdlist,1,-1)
+vfats = []
+
+if args.vfat=="none":
+    print("Please pass the name of the vfat(s) you want to analyse with the option --vfat")
+    sys.exit()
+elif args.vfat=="all":
+    for x in range(0,24):
+        vfats.append(str(x))
+else:
+    for vf in (args.vfat).split(","):
+        vfats.append(vf)
+#print vfats
```

Comment:
I would have this do all vfats by default; there's probably no case where we would be interested in just *one* VFAT.  The DAC scan application doesn't even allow for a single VFAT scan.

Reply:
I thought the macro to perform a quick check on a single VFAT that shows problems during the different scandates. It is flexible to perform in one time also all the VFATs, by creating different files for each VFAT.

Reply:
Yes I understand; but I'm asking to expand the scope to something with a broad scope of validity.

Reply:
I will set the default to all and leave also the possibility to check a single VFAT. Is it fine?

Reply:
Sounds great! Thanks